src/crafting_gui.cpp

```diff
@@ -910,34 +887,64 @@ static bool mouse_in_window( cata::optional<point> coord, const catacurses::wind
     return false;
 }
 
-static void perform_nested( const recipe *rec, std::string filterstring, tab_list &tab,
-                            tab_list &subtab, bool &recalc )
+static void recursively_expance_recipes( std::vector<const recipe *> &current,
+        std::vector<int> &indent, std::map<const recipe *, availability> &availability_cache, int i,
+        Character &player_character, bool unread_recipes_first, bool highlight_unread_recipes )
 {
-    // if we are already looking at a nested category then we should spread tabs from where we are
-    int index = nested_index( tab, subtab );
-    if( index != -1 ) {
-        //delete categories past where we are
-        uistate.nested_recipes.resize( index + 1 );
-    } else {
-        uistate.nested_recipes.clear();
+    std::vector<const recipe *> tmp;
+    for( const recipe_id nested : current[i]->nested_category_data ) {
```

Comment:
```suggestion
    for( const recipe_id &nested : current[i]->nested_category_data ) {
```
Clang 12 was complaining about this one